File: C++/题目1108：堆栈的使用.cpp

#include <stdio.h>
#include <stack>   
#include <string.h>
#include <string>
using namespace std; 
int n,i; 

int main(){
    while(scanf("%d",&n) != EOF){
		if(n == 0){
			break;
		}
        stack<int> numStack;
		for(i = 0; i < n; i++){
			char operate[2];
			scanf("%s",operate);
			if(operate[0] == 'P'){
				int k;
				scanf("%d",&k);
				numStack.push(k);
			}else if(operate[0] == 'O'){
				if(!numStack.empty()){
					numStack.pop();
				}
			}else{
				if(numStack.empty()){
					printf("E\n");
				}else{
					printf("%d\n",numStack.top());
				}
			}
		}
		printf("\n");
    }
    return 0;
} 

/**************************************************************
	Problem: 1108
	User: wangzhenqing
	Language: C++
	Result: Accepted
	Time:20 ms
	Memory:1052 kb
****************************************************************/


                        

File: Java/题目1108：堆栈的使用.java

// 题目1108：堆栈的使用

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:53
 * @url:http://ac.jobdu.com/problem.php?pid=1108
 */

import java.util.Scanner;
import java.util.Stack;

public class Main {
    /*
     * 1108
     */
public static void main(String[] args) throws Exception{
		Scanner scanner = new Scanner(System.in);
		while (scanner.hasNextLine()) {
			String n = scanner.nextLine().trim();
			int numN = Integer.parseInt(n);
			if (numN == 0 ) {
				break;
			}
			Stack<Integer> stack = new Stack<Integer>();
			for (int i = 0; i < numN; i++) {
				String a = scanner.nextLine().trim();
				if (a.equals("A")) {
					if (stack.isEmpty()) {
						System.out.println("E");
					}else {
						System.out.println(stack.peek());
					}
					
				}
				
				if (a.equals("O")) {
					if (!stack.isEmpty()) {
						stack.pop();
					}
				}
				
				if (a.contains("P")) {
					int tempNum = Integer.parseInt(a.substring(2));
					stack.push(tempNum);
				}
			}
			System.out.println();
		}
	}
}
/**************************************************************
 Problem: 1108
 User: wzqwsrf
 Language: Java
 Result: Accepted
 Time:1050 ms
 Memory:104080 kb
 ****************************************************************/




